src/frontend/src/expr/agg_call.rs

```diff
@@ -93,7 +93,10 @@ impl AggCall {
                 Float32 | Float64 | Int256 => Float64,
                 _ => return Err(err()),
             },
-            (AggKind::PercentileCont, _) => Float64,
+            (AggKind::PercentileCont, [input]) => match input {
+                Float64 => Float64,
+                _ => return Err(err()),
+            },
```

Comment:
Since `percentile_*` and `mode` are so called _ordered-set aggregate functions_, I'm thinking that it'll be better to have a separate `infer_return_type_for_ordered_set_agg(agg_kind, inputs, direct_args)`, and leave `infer_return_type` to handle only normal agg calls, both panicing on the other kinds.

Also, it'll be sweet to have `AggKind::is_ordered_set` and use it in relevant places to increase readibility.